bibtex-actions.el

```diff
@@ -0,0 +1,116 @@
+;;; bibtex-actions.el --- Bibliography manager action
+
+;;; Code:
+
+(require 'bibtex-completion)
+
+(defvar bibtex-actions-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map (kbd "p") 'bibtex-actions-open-pdf)
+    (define-key map (kbd "u") 'bibtex-actions-open-url-or-doi)
+    (define-key map (kbd "c") 'bibtex-actions-insert-citation)
+    (define-key map (kbd "r") 'bibtex-actions-insert-reference)
+    (define-key map (kbd "k") 'bibtex-actions-insert-key)
+    (define-key map (kbd "b") 'bibtex-actions-insert-bibtex)
+    (define-key map (kbd "a") 'bibtex-actions-add-PDF-attachment)
+    (define-key map (kbd "e") 'bibtex-actions-edit-notes)
+    (define-key map (kbd "s") 'bibtex-actions-show-entry)
+    (define-key map (kbd "l") 'bibtex-actions-add-pdf-to-library)
+    map))
+
+(defun bibtex-actions--get-candidates ()
+  "Return all keys from bibtex-completion-candidates."
+  (mapcar
+   (lambda (cand)
+     (cons (bibtex-completion-format-entry cand (1- (frame-width)))
```

Comment:
We need a better solution for this, as we're missing data in the process.

If we can't complete against the output of `bibtex-completion-candidates` using `completing-read`, can we transform it so we don't lose any of that data?

The use case here is I really want to add tags to complete against, so I can distinguish among different projects, etc.

See #362.

Reply:
I posted the [question on reddit](https://www.reddit.com/r/emacs/comments/lsxjfx/completingread_displaying_using_one_string_but/), and got a response from one of the selectrum devs.

Seems we want to use propertize something like this.

``` elisp
(completing-read ";) "
                 (list (concat "vim" (propertize "there is evil-mode" 'invisible t))
                       (concat "emacs"  (propertize "the best" 'invisible t))))
```

So in other words, if `bibtex-completion-additional-search-fields` is set, add those fields from `bibtex-completion-candidates`, but hide them.

I guess this is roughly similar to the ivy display transformer.